lora/models.py

```diff
@@ -47,7 +48,10 @@ def __init__(
         self.lora_b = mx.zeros(shape=(lora_rank, output_dims))
 
     def __call__(self, x):
-        y = self.linear(x.astype(self.linear.weight.dtype))
+        if isinstance(self.linear, nn.QuantizedLinear):
```

Comment:
Why not cast in the case of `nn.QuantizedLinear` too?
```python
dtype = self.linear.weight.dtype
if isinstance(self.linear, nn.QuantizedLinear):
    dtype = self.linear.scales.dtype
y = self.linear(x.astype(dtype))
```

Reply:
Yes, I like that. Thank you1